druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -514,12 +515,14 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
             final String origin = druidTable.getRowType(getCluster().getTypeFactory())
                 .getFieldList().get(ref.getIndex()).getName();
             if (origin.equals(druidTable.timestampFieldName)) {
-              granularity = Granularity.NONE;
-              builder.add(s);
+              granularity = Granularity.ALL;
+              timeExtractionDimensionSpec = TimeExtractionDimensionSpec.makeFullTimeExtract();
+              dimensions.add(timeExtractionDimensionSpec);
+              builder.add(DruidConnectionImpl.DEFAULT_RESPONSE_TIMESTAMP_COLUMN);
```

Comment:
if we use a custom name let say `timeColumn` will need to instruct druid to send results back with that name. Does that answer to your question ?